src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -238,14 +287,10 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 		$options = wp_parse_args( $options, array( 'with_supports' => true ) );
 
 		if ( null === static::$theme || ! static::has_same_registered_blocks( 'theme' ) ) {
-			$wp_theme        = wp_get_theme();
-			$theme_json_file = $wp_theme->get_file_path( 'theme.json' );
-			if ( is_readable( $theme_json_file ) ) {
-				$theme_json_data = static::read_json_file( $theme_json_file );
-				$theme_json_data = static::translate( $theme_json_data, $wp_theme->get( 'TextDomain' ) );
-			} else {
-				$theme_json_data = array();
-			}
+			$wp_theme = wp_get_theme();
+
+			// Read main theme.json (which may also come from the parent theme).
+			$raw_theme_json_data = static::read_theme_json_data_for_theme( $wp_theme );
```

Comment:
Note that this change here actually includes a bug fix for something I just noticed while working on it:
* `WP_Theme::get_file_path()` may return the parent theme's `theme.json` file.
* But then below it is translated using the _child theme's_ text domain, which will lead to translations not to work correctly given the text domain is probably different for the parent and child theme.

The way I have updated the code now this is no longer a problem.